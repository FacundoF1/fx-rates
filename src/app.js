'use strict';

const Inert = require('@hapi/inert');
const Vision = require('@hapi/vision');
const HapiSwagger = require('hapi-swagger');

const server = require('../config/server');
const baseRouter = require('./routes');

const init = async () => {

    const swaggerOptions = {
        "info": {
            "title": "Swagger - FX Rates",
            "description": "Generated with documentation dinamyc created by Facundo Ferrari.",
            "version": "1.0.0"
        },
        schemes: ['http']
    }

    // Adding plugins for swagger docs;
    await server.register([
        Inert,
        Vision,
        {
            plugin:HapiSwagger,
            options:swaggerOptions
        }
    ])

    await server.register(baseRouter,{
		routes:{
			prefix:'/v1'
		}
	});

    server.events.on('response', function (request) {
        console.log(request.info.remoteAddress + ': ' + request.method.toUpperCase() + ' ' + request.path + ' --> ' + request.response.statusCode);
    });

    await server.start();
    console.log('Server running on %s', server.info.uri);
};

process.on('unhandledRejection', (err) => {
    console.log(err);
    process.exit(1);
});

init();