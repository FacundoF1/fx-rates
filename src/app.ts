import express from 'express';
import logger from 'morgan';
import routes from './routes';
import { error404Handler, errorHandler } from './middleware/index';

const app = express();

const path = require('path');
const swaggerUi = require('swagger-ui-express');
const YAML = require('yamljs');
const swaggerDocument = YAML.load('docs/specs/swagger.yaml');


app.use(express.json());
app.use(express.static(path.join(__dirname)));


app.use(logger('dev', {}));
app.use(express.json());
app.use(express.urlencoded({ extended: false }));

app.use('/api-docs', swaggerUi.serve, swaggerUi.setup(swaggerDocument));
app.use('/v1', routes);

app.use(error404Handler);
app.use(errorHandler);

export { app };
