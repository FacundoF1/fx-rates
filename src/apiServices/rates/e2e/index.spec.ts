import { app } from '../../../app';
import supertest from 'supertest';
const request = supertest.agent(app);

describe('Rates currency fee amount', () => {
    it('Should response amount more rates with fee', () => {
        return request
            .post('/v1/rates/EURBRL')
            .send({ 'amount': 25 })
            .expect(200)
            .then(response => expect(response.body).toMatchSnapshot());
    });
    it('Create rate fee', () => {
        return request
            .post('/v1/rates/fee')
            .send({
                pair: "EURBRL",
                porcent_fee: "5"
            })
            .expect(200)
            .then(response => expect(response.body).toMatchSnapshot());
    });
});