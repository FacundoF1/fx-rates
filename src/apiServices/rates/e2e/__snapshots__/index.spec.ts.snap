// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Rates currency fee amount Create rate fee 1`] = `
Object {
  "_id": "kn3ULBTplUrDp2Og",
  "fee": 5,
  "pair": "EURBRL",
}
`;

exports[`Rates currency fee amount Should response amount more rates with fee 1`] = `
Object {
  "fee": "5 %",
  "fee_amount": 145.77633,
  "original_rate": 5.553384,
  "pair": "EURBRL",
  "rate": 5.8310532,
}
`;
